puppet-mode.el

```diff
@@ -632,6 +633,40 @@ of the initial include plus puppet-include-indent."
         (indent-line-to 0)))))
 
 
+;;;; SMIE
+
+(defvar puppet-use-smie nil
+  "Whether to use SMIE or not.")
+
+(defvar puppet-smie-verbose nil)
+
+(defun puppet--smie-forward-token ()
+  (smie-default-forward-token))
+
+(defun puppet--smie-backward-token ()
+  (smie-default-backward-token))
+
+(defconst puppet-smie-grammar
+  (smie-prec2->grammar
+   (smie-bnf->prec2
+    '((id)))))
+
+(defun puppet-smie-rules (kind token)
+  "Indentation rules for Puppet."
+  (when puppet-smie-verbose
+    (save-excursion
+      (let ((prev (smie-indent-backward-token))
+            (next (smie-indent-forward-token)))
+        (message "kind=%S token=%S prev=%S next=%S parent=%S"
```

Comment:
Yes, but it's still just redundant debugging code which serves no specific purpose by itself, and also only covers the particular instance of debugging that you needed in the past.  But it's of no use generally, so please remove it and the variable as well.  

If anyone needs to debug this particular place again, they can easily add in their own debugging code again and press `C-M-x`.